src/sqlstorage.cc

```diff
@@ -16,7 +17,18 @@ SQLStorage::SQLStorage(const StorageConfig& config) : INvStorage(config) {
                              " missing");
   }
 
-  boost::filesystem::create_directories(config.sqldb_path.parent_path());
+  if (!boost::filesystem::is_directory(config.sqldb_path.parent_path())) {
+    boost::filesystem::create_directories(config.sqldb_path.parent_path());
```

Comment:
Is there not a way to create a directory and set the permissions all at once? Baiscally `mkdir -m 700`.